.github/workflows/ci.yml

```diff
@@ -122,7 +123,7 @@ jobs:
 
   auto-approve-staging:
     runs-on: ubuntu-latest
-    if: startsWith(github.ref, 'refs/tags/v')
+    if: ${{ needs.semantic.outputs.sem_version_bool == 'true' }}
```

Comment:
Since `auto-approve-staging` is a different job, it needs to reference the originating job from where the output var. See here for an example:
https://github.com/hirosystems/explorer/blob/72b71f3f0fc6272d6f0ab4a180c738cbde1b58f9/.github/workflows/ci.yml#L120

Also, brackets aren't needed for the `if` property.